source/ovcrptvw.pas

```diff
@@ -5627,29 +5627,20 @@ function TOvcCustomReportView.CalcHLinesNet: Integer;
 
 function TOvcCustomReportView.CalcHPages(const SelectedOnly : Boolean; const LinesPerPage: Integer): Integer;
 var
-  DoneSectionHeader, DoneSectionFooter: Boolean;
   j, LinesLeft: Integer;
 begin
   Result := 0;
   LinesLeft := 0;
-  DoneSectionHeader := False;
-  DoneSectionFooter := False;
   for j := 0 to pred(Lines) do
     if not SelectedOnly or IsSelected[j] then begin
       while LinesLeft <= 0 do begin
         inc(LinesLeft, (LinesPerPage - PrinterProperties.PrintHeaderLines - PrinterProperties.PrintFooterLines));
         inc(Result);
       end;
-      if not DoneSectionHeader then begin
         if CurrentView.ShowHeader then
           dec(LinesLeft, 2);
-        DoneSectionHeader := True;
-      end;
-      if not DoneSectionFooter then begin
         if CurrentView.ShowFooter then
           dec(LinesLeft, 2);
-        DoneSectionFooter := True;
-      end;

```

Comment:
At a quick glance it appears to me that the code you are removing might actually work if `DoneSectionHeader` and `DoneSectionFooter` were initialized to `false` at the top of the `for j` loop like so:
```pascal
function TOvcCustomReportView.CalcHPages(const SelectedOnly : Boolean; const LinesPerPage: Integer): Integer;
var
  DoneSectionHeader, DoneSectionFooter: Boolean;
  j, LinesLeft: Integer;
begin
  Result := 0;
  LinesLeft := 0;
  for j := 0 to pred(Lines) do
  begin
    DoneSectionHeader := False;
    DoneSectionFooter := False;
    if not SelectedOnly or IsSelected[j] then
    begin
      while LinesLeft <= 0 do
      begin
        inc(LinesLeft, (LinesPerPage - PrinterProperties.PrintHeaderLines - PrinterProperties.PrintFooterLines));
        inc(Result);
      end;
      if not DoneSectionHeader then
      begin
        if CurrentView.ShowHeader then
          dec(LinesLeft, 2);
        DoneSectionHeader := True;
      end;
      if not DoneSectionFooter then
      begin
        if CurrentView.ShowFooter then
          dec(LinesLeft, 2);
        DoneSectionFooter := True;
      end;
      dec(LinesLeft);
      if PrintDetailView <> nil then
      begin
        DoDetail(ItemData[j]);
        dec(LinesLeft, PrintDetailView.CalcHLinesNet);
      end;
    end;
  end;
end;
```

See if my suggestion remedies the issue you have been facing?  I can't tell what the relevance of the two booleans is, they seem redundant.  Also the `while loop` I think could probably just be an `if` statement.  But at any rate I would be curious to see if a less drastic change to this code can yield just as good a result as what you  are proposing.  I moved a couple `begin` statements.  You don't have to duplicate that aspect of my change.  I don't care for begins to start at the end of a line.  Makes the code more difficult to read in my opinion.

Reply:
Interestingly, your suggestion is exactly what I initially did to fix this bug, but when I looked at the code more closely it seemed extremely insufficient to set those booleans to false every time in that for loop just so that they can be set to true a few lines lower.

The original Booleans' purpose is to limit the subtraction of the header and footer lines only once per report, but this is clearly wrong logic as the header and footer appears on every page not just at the beginning and end of the report.

I don't think the while loop can be changed to an if statement, but didn't analyze this in any detail.

Reply:
The more I stare at the original code the more the original code looks like it should work fine...  Can you duplicate the issue you are having in a simple project that you could share in a repo.  If you want to make that repo private and add me as a collaborator I would be willing to take a look.  I have not used report view since the early 2000's and back then I don't think I ever tried printing.